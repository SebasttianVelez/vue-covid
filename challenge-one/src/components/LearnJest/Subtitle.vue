<template>
  <div class="title">
    <h1>{{ subtitle }}</h1>
  </div>
</template>

<script>
export default {
  name: 'Subtitle',
  props: ['subtitle'],
}
</script>

<style></style>
