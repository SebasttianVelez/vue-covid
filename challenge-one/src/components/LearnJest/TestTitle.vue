<template>
  <div class="container">
    <h1>Este es un titulo: {{ title }}</h1>
    <Subtitle :subtitle="subtitle" />
    <p>
      Ullamco duis aute ut aute. Pariatur et magna aliqua culpa aute Lorem id
      non velit sit culpa consequat. Ullamco non pariatur cillum sunt excepteur
      eiusmod nisi et ipsum dolore non qui officia.
    </p>
  </div>
</template>

<script>
import Subtitle from './Subtitle.vue'
export default {
  props: ['title'],
  components: {
    Subtitle,
  },
  data() {
    return {
      subtitle: 'SUBTITLE SUCCESS',
    }
  },
}
</script>

<style></style>
