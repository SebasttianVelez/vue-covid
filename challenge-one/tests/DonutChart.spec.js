import TestTitle from '../src/components/LearnJest/TestTitle.vue'
import Subtitle from '../src/components/LearnJest/Subtitle.vue'
import { mount, shallowMount } from '@vue/test-utils'

describe('TestTitle', () => {
  // Comprobar si renderiza el componente
  it('it renders the component', () => {
    const wrapper = shallowMount(TestTitle)
    expect(wrapper.html()).toMatchSnapshot()
  })

  // Comprobar si un componente contiene el dato title
  it('Should mount the component and contain title', () => {
    const title = ''
    const testTitle = mount(TestTitle, {
      propsData: {
        title: title,
      },
    })
    expect(testTitle.text()).toContain(title)
  })

  //componentes dentro de otros
  it('it renders the right', () => {
    // arrange
    const wrapper = shallowMount(TestTitle)
    const subtitle = wrapper.findComponent(Subtitle)

    // assert
    expect(subtitle.exists()).toBe(true)
  })
})
