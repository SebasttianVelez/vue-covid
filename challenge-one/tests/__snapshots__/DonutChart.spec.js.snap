// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestTitle it renders the component 1`] = `
"<div class=\\"container\\">
  <h1>Este es un titulo: </h1>
  <subtitle-stub subtitle=\\"SUBTITLE SUCCESS\\"></subtitle-stub>
  <p>
    Ullamco duis aute ut aute. Pariatur et magna aliqua culpa aute Lorem id
    non velit sit culpa consequat. Ullamco non pariatur cillum sunt excepteur
    eiusmod nisi et ipsum dolore non qui officia.
  </p>
</div>"
`;
